modules/generator/processor/servicegraphs/servicegraphs.go

```diff
@@ -301,12 +324,22 @@ func (p *Processor) onExpire(e *store.Edge) {
 		// We check if the span we have is the root span, and if so, we set the client service to "user".
 		if _, parentSpan := parseKey(e.Key()); len(parentSpan) == 0 {
 			e.ClientService = "user"
+
+			if p.Cfg.EnableVirtualNodeLabel {
+				e.Dimensions["virtual_node"] = "client"
+			}
```

Comment:
does this add any new info that just checking the client service label wouldn't have otherwise? i.e. if clientservice = user then we know what value virtual node would have had?

Reply:
at the moment no, but I'd vouch for adding it anyway because:

1. label consistency for the new label: since adding `server` adds new info, by not adding `client` too we can end up in a strange situation where you need a different query for filtering (something like `{__virtual_node: server}` vs. `{__virtual_node: "", client: user}`)
2. it looks like the client service label could be improved in the future as well (the current value of `user` may not be accurate for all cases, but a best guess)